nifi-nar-bundles/nifi-soap-bundle/nifi-soap-processors/src/main/java/org/apache/nifi/processors/soap/GetSOAP.java

```diff
@@ -152,20 +152,75 @@
             .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
             .build();
 
+    protected static final PropertyDescriptor AUTH_BY_HEADER = new PropertyDescriptor.Builder()
+            .name("HeaderAuthentication")
+            .description("If you need to do authentication with SOAP headers, this must be a json string " +
+                         "defining the structure since it varies from service to service. Username and Password " +
+                         "must be set in the separate parameters to this processors and referenced here " +
+                         "as variables ${Username} and ${Password}. For example, " +
+                         "{ \"UserAuthentication\": { \"username\": \"${Username}\", \"password\": \"${Password}\"," +
+                         " \"some_other_custom_field\": \"value\" }")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(new Validator() {
+                @Override
+                public ValidationResult validate(String subject, String value, ValidationContext context) {
+                    try {
+                        return (new ValidationResult.Builder())
+                                .subject(subject)
+                                .input(value)
+                                .explanation("Header Authentication must be empty or valid json")
+                                .valid("".equals(value) || mapper.readTree(value) != null).build();
+                    }
+                    catch (IOException e) {
+                        return (new ValidationResult.Builder())
+                                .subject(subject)
+                                .input(value)
+                                .explanation("Header Authentication must be empty or valid json: " + e.getMessage())
+                                .valid(false).build();
+                    }
+                }
+            })
+            .build();
+
+    protected static final PropertyDescriptor API_NAMESPACE = new PropertyDescriptor.Builder()
+            .name("Namespace")
+            .description("XML Namespace.")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dynamic(false)
+            .build();
+
+    protected static final PropertyDescriptor SKIP_FIRST_ELEMENT = new PropertyDescriptor.Builder()
+            .name("SkipFirstElement")
+            .description("Return the XML that comes after the first element.")
+            .required(true)
+            .defaultValue("false")
+            .allowableValues("true", "false")
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .dynamic(false)
+            .build();
 
```

Comment:
For all properties (ENDPOINT_URL, WSDL_URL, API_NAMESPACE,USER-AGENT, SKIP_FIRST_ELEMENT,etc.) would we want to support EL? Then we can have an incoming FF that has attributes that can be used to populate them.

Also, current best practice is to have name and displayName, such as:
	    .name("endpoint-url")
            .displayName("Endpoint URL")